### PR TITLE
fix: Tweak AMVF config with time in Examples

### PR DESCRIPTION
**Pull request by andiwand**
I observed higher vertex efficiency with time setting `maxMergeVertexSignificance` to default. `tracksMaxSignificance` and `doFullSplitting` were also reset in order to use the same config with and without time.

### CODE REVIEW

**Comment by andiwand** [comment]
Let's see what the physmon performance says

**Review by paulgessinger — APPROVED** [review]
Physmon seems compatible with efficiency increase.

**Comment by andiwand** [comment]
After discussion with @pbutti we agreed on splitting the change to see what option drives what performance.

One problem with our current monitoring is that we do not know if the additional vertices are duplicates or fakes. We will have to dig a bit deeper to check that.

**Review by felix-russo — DISMISSED** [review]
lgtm

For lower pile up (i.e., mu=50 like in our physmon), I used to see a better performance when this line was there

For higher pile-up  (i.e., mu = 200 like in the setup for the paper) the performance was better without it

If now the performance is better for low pile-ups as well, we should definitely remove it 

If we encounter bad performance in a new setup, I would always consider tweaking this parameter too see what happens. Adding a python binding might be good in the future.

**Comment by andiwand** [comment]
physmon looks good now

File: Examples/Algorithms/Vertexing/src/AdaptiveMultiVertexFinderAlgorithm.cpp [hunk 1]
```diff
@@ -133,17 +133,18 @@ auto ActsExamples::AdaptiveMultiVertexFinderAlgorithm::makeVertexFinder() const
   finderConfig.tracksMaxZinterval = 1. * Acts::UnitConstants::mm;
   finderConfig.maxIterations = 200;
   finderConfig.useTime = m_cfg.useTime;
+  finderConfig.tracksMaxSignificance = 5;
+  // This should be used consistently with and without time
+  finderConfig.doFullSplitting = false;
+  finderConfig.maxMergeVertexSignificance = 3;
```
Comment (andiwand) [inline Examples/Algorithms/Vertexing/src/AdaptiveMultiVertexFinderAlgorithm.cpp:139]:
```suggestion
  // corresponds to a p-value of 0.92 using `chi2(x=3,ndf=1)`
  finderConfig.maxMergeVertexSignificance = 3;
```

File: Examples/Algorithms/Vertexing/src/AdaptiveMultiVertexFinderAlgorithm.cpp [hunk 2]
```diff
@@ -133,17 +133,18 @@ auto ActsExamples::AdaptiveMultiVertexFinderAlgorithm::makeVertexFinder() const
   finderConfig.tracksMaxZinterval = 1. * Acts::UnitConstants::mm;
   finderConfig.maxIterations = 200;
   finderConfig.useTime = m_cfg.useTime;
+  finderConfig.tracksMaxSignificance = 5;
```
Comment (andiwand) [inline Examples/Algorithms/Vertexing/src/AdaptiveMultiVertexFinderAlgorithm.cpp:136]:
```suggestion
  // corresponds to a p-value of 0.92 using `chi2(x=5,ndf=2)`
  finderConfig.tracksMaxSignificance = 5;
```

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [dac4a44d21d5cebbe402d4c2ab1a6c78c02c892b]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026936)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026935)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026934)||
|:green_circle: | [test_ActsConversionWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026933)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026932)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026931)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026930)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026929)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026928)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026927)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026926)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026925)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026924)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026923)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026922)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026921)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026920)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026919)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026918)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026917)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026916)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026915)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026914)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026913)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/37026912)||

**Pull request merged**